File: biweekly/095/q3.cc
class Solution {
public:
    int xorBeauty(vector<int>& nums) {
        int n = nums.size();
        int an = 0;
        for (auto& a : nums) an ^= a;
        return an;
        
        unordered_map<int, int> t1, m;
        /*
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {*/
        for (auto& a : nums) {
            for (auto& b : nums) {
                ++t1[a | b];
            }
        }
        
        for (auto& a : nums) {
            for (auto& [val, cnt] : t1) {
                if (cnt % 2 == 0) continue;
                m[a | val] += cnt;
            }
        }
        
        int ans = 0;
        for (auto& [val, cnt] : m) {
            if (cnt & 1) ans ^= val;
        }
        return ans;
    }
};

File: weekly/328/q3.py
class Solution:
    def countGood(self, nums: List[int], k: int) -> int:
        cnt = Counter()
        ans = left = pairs = 0
        for x in nums:
            pairs += cnt[x]
            cnt[x] += 1
            ans += left
            while pairs >= k:
                ans += 1
                cnt[nums[left]] -= 1
                pairs -= cnt[nums[left]]
                left += 1
        return ans
